api/receiver/appengine.go

```diff
@@ -57,12 +58,24 @@ func NewAppEngineAPI(ctx context.Context) AppEngineAPI {
 	return &appEngineAPIImpl{
 		AppEngineAPIImpl: shared.NewAppEngineAPI(ctx),
 		queue:            ResultsQueue,
+		store:            shared.NewAppEngineDatastore(ctx, false),
 	}
 }
 
 func (a *appEngineAPIImpl) addTestRun(testRun *shared.TestRun) (*DatastoreKey, error) {
-	key := datastore.NewIncompleteKey(a.Context(), "TestRun", nil)
-	key, err := datastore.Put(a.Context(), key, testRun)
+	ctx := a.Context()
+	var key *datastore.Key
+	if testRun.ID > 0 {
```

Comment:
```suggestion
	if testRun.ID != 0 {
```

I think the ID *could* be negative although I've never seen one. And all the documentation only says 0 is the special case, so let's be a bit more cautious here.